scipy/stats/tests/test_distributions.py

```diff
@@ -1414,20 +1414,32 @@ def test_mean_small_p(self):
 
 
 class TestGumbel_r_l:
-    def setup_method(self):
-        np.random.seed(1234)
+    @pytest.fixture(scope='function')
+    def rng(self):
+        return np.random.default_rng(1234)
 
     @pytest.mark.parametrize("dist", [stats.gumbel_r, stats.gumbel_l])
-    @pytest.mark.parametrize("loc_rvs,scale_rvs", ([np.random.rand(2)]))
-    def test_fit_comp_optimizer(self, dist, loc_rvs, scale_rvs):
-        data = dist.rvs(size=100, loc=loc_rvs, scale=scale_rvs)
+    @pytest.mark.parametrize("loc_rvs", [-1, 0, 1])
+    @pytest.mark.parametrize("scale_rvs", [.1, 1, 5])
+    @pytest.mark.parametrize('fix_loc, fix_scale',
+                             ([True, False], [False, True]))
+    def test_fit_comp_optimizer(self, dist, loc_rvs, scale_rvs,
+                                fix_loc, fix_scale, rng):
+        data = dist.rvs(size=100, loc=loc_rvs, scale=scale_rvs,
+                        random_state=rng)
 
         # obtain objective function to compare results of the fit methods
         args = [data, (dist._fitstart(data),)]
         func = dist._reduce_func(args, {})[1]
 
+        kwds = dict()
+        if fix_loc:
+            kwds['floc'] = loc_rvs * 2
```

Comment:
Maybe comment - the value `* 2` is arbitrary, but you're making sure that fixed value is not even close to the true value?

Reply:
Correct